che-tomcat8-slf4j-logback/src/assembly/conf/tomcat-json-appenders.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2012-2018 Red Hat, Inc.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+    Contributors:
+      Red Hat, Inc. - initial API and implementation
+
+-->
+<included>
+    <appender name="stdout-json" class="org.apache.juli.logging.ch.qos.logback.core.ConsoleAppender">
+        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
+            <includeMdcKeyName>identity_id</includeMdcKeyName>
```

Comment:
isn't that specific to rh-che ?
if we include these mdc property in json, why not in plain text ?

Reply:
> isn't that specific to rh-che ?

No, there are bindings [here](https://github.com/garagatyi/che/blob/06b32260dbce4637e547ebea61b63b8848e9052a/assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterServletModule.java#L47) and [here](https://github.com/garagatyi/che/blob/db0644505c1ed8936d57677412161f0d86bc2b58/multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/deploy/KeycloakServletModule.java#L32) of filters that set `req_id` and `identity_id` in MDC context in upstream Che.

> if we include these mdc property in json, why not in plain text ? 

It wasn't needed previously in upstream Che, so I'm not sure we should add it now without any need. In addition, it would change the look of upstream Che logs, but I'm not sure devs want it, at least right now. But in case of JSON RhChe can benefit from it.

Reply:
To me, it's not because we have bound filters that can take care of some mdc properties that we should display all of them or add them to the configuration. To me displaying MDC is up to any deployment and production policy. . Should these filters be bound by default in Che ? Are these properties useful for any deployment ?

But if we decide to show in json identity_id and request_id because you think it will be useful in any Eclipse Che deployment, then we should also display these MDC properties in plain text.
So to me in upstream project, either we display identity_id and request_id in both json and plain text. either we don't display them at all.

That makes me think that it may not a good way to handle the issue: the solution should be more generic than just switching from plain text to json: it should be about switching the output logger from one logger configuration to another (whether it is json or plaintext or json_rhche or anything else)

In the logging world every ops team will have their preferences to show logs in a particular format and which information to show. And we are only talking about the output stream.

Reply:
PS: 
OK just seen that we can add custom appenders, so don't take my generic remarks as it is already generic ;)

Reply:
Yes, one can provide a custom loggers configuration if ops team needs this. And switching would require an addition of 2 files and change an environment variable value. Unlike current solution in che-lib, one won't need to disable/remove not needed default loggers.

> Can't we assume that json is for scenarios where logs are parsed by non-human application (and more detailed info make sense for deeper cross linking) whereas text logs are for scenarios where logs are read by humans (hence less details is better)?

Me, as a developer, definitely don't want to see longer logs entries in development unless I'm investigation a weird issue that depends on interactions with other our services. But, fortunately,  this rarely happens. I would leave default behavior in the same state as it was before my changes because it is convenient for development.

Reply:
OK to merge as it is and create a new issue to discuss about adding MDC `identity_id` and `request_id` in plaintext as well https://github.com/eclipse/che/issues/9334